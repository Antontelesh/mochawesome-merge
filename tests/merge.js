const { merge } = require('../lib')

describe('merge', () => {
  test('merges configs', async () => {
    const report = await merge({
      files: [`./tests/mochawesome-report/mochawesome*.json`],
    })
    const suites = report.results

    expect(report.stats).toMatchSnapshot({
      start: expect.any(String),
      end: expect.any(String),
    })
    expect(suites.length).toBe(3)

    expect(suites[0].tests.length).toBe(1)
    expect(suites[0].passes.length).toBe(0)
    expect(suites[0].failures.length).toBe(1)

    expect(suites[1].tests.length).toBe(3)
    expect(suites[1].passes.length).toBe(1)
    expect(suites[1].failures.length).toBe(2)

    expect(suites[2].tests.length).toBe(2)
    expect(suites[2].passes.length).toBe(2)
    expect(suites[2].failures.length).toBe(0)
  })

  test('throws when invalid directory provided', async () => {
    const reportsGlob = './invalid-directory/mochawesome*.json'

    await expect(merge({ files: [reportsGlob] })).rejects.toEqual(
      new Error(`Pattern ${reportsGlob} matched no report files`)
    )
  })

  test('defaults to mochawesome-report directory', async () => {
    const reportsGlob = './mochawesome-report/mochawesome*.json'
    await expect(merge()).rejects.toEqual(
      new Error(`Pattern ${reportsGlob} matched no report files`)
    )
  })
})
