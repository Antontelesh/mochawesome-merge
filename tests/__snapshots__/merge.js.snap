// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`merge merges configs 1`] = `
Object {
  "duration": 600,
  "end": Any<String>,
  "failures": 3,
  "hasOther": false,
  "hasSkipped": false,
  "other": 0,
  "passPercent": 50,
  "passes": 3,
  "pending": 0,
  "pendingPercent": 0,
  "skipped": 0,
  "start": Any<String>,
  "suites": 3,
  "tests": 6,
  "testsRegistered": 6,
}
`;
